File: lua/skeleton.lua
#!/usr/bin/lua

--This is our subroutine at the beggining of script
function check_length(value)
	if string.len(value) > 30 then	--if the string was bigger than 30 chars, FAIL
		print("You didn't follow the rules, enter a line that is less than 30 characters:\n")
		return "FAIL"
	end
end

print("Welcome to our example Lua script")
print("please enter your file name: ")
local filename = io.read()  --read user input for filename

local FILE = (io.open(filename, "r"))	--Open a filehandle
local line_amount = 0					--init variable for amount of lines
local line_length = {}					--init an array for length of each line
local last								--will contain our last line
while true do							--infinite while loop
	local line = FILE:read("*line")		--read current line of file
	if not line then break end			--exit loop if no more lines
	last = line 						--note the current line (last loop will have last line)
	line_length[line_amount] = string.len(line)	--comput length of current line and store to array
	line_amount = line_amount + 1		--Note the increase of lines/loops
end
FILE:close()	--close our filehandle

print("Our last line was\n" .. last)	--display what our last line was

local i = 0					--loop iterator init
local total_chars = 0		--set total characters to 0
while line_amount > i do	--while we still have lines to process
	total_chars = total_chars + line_length[i]	--tally the total characters from line
	i = i + 1				--increment loop counter
end

--print off some general statisics on file
print("The file had " .. line_amount .. " lines and " .. total_chars .. " total characters\n")

print("Enter another line that is less than 30 characters:\n") --prompt for another line
local another_line = io.read()		--see if it is more than 30 characters
while check_length(another_line) == "FAIL" do	--keep checking that it is < 30 chars
	another_line = io.read()		--see if it is more than 30 characters
end

local FILE = (io.open(filename, "a"))	--re-open file in append mode
FILE:write(another_line .. "\n")		--write to file

File: ruby/skeleton.rb
#!/usr/bin/ruby

#Define our subroutine for checking line length (in characters)
def check_length value
	if value.length > 30
		puts "You didn't follow the rules, enter a line that is less than 30 characters:\n"
		return "FAIL"
	else
		return
	end
end

puts "Welcome to our example ruby script\n"
puts "please enter your file name: ";
filename = gets.chomp						#Get input from user
puts "The filename is %s" % filename		#This is variable interpolation

file = File.open(filename, "r")		#how to read a file
last_line = ""						#init the variable
line_count = 0						#init linecount to 0
line_length = Array.new				#how to init an array
file.each_line do |line|			#go through each line, will be stored in var "line"
	last_line = line 						#note the current line in last_line
	line_length[line_count] = line.length 	#count chars and put it in our array
	line_count = line_count + 1 			#Iterate the line counter
end
file.close									#close our file up

puts "The last line was %s" % last_line		#report what the last line was

i = 0								#set loop iterator
total_chars = 0						#init total characters to 0
while line_count > i  do			#while we still have lines
	total_chars += line_length[i]	#tally up the character count
	i += 1 						    #next
end

puts "The file had %s lines and %s total characters\n" % [line_count, total_chars] #stats

puts "Enter another line that is less than 30 characters:\n"
next_line = gets.chomp				#get input from user
check_length (next_line)			#user our subroutine to check length
while check_length (next_line) == "FAIL"	#keep checking that it is < 30 chars
	next_line = gets.chomp
end

file = File.open(filename, "a") 	#"a" for append
	file.write(next_line + "\n")	#write user submitted line to file
file.close
